miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/permission/MiniAppCustomPermissionCache.kt

```diff
@@ -37,41 +37,12 @@ internal class MiniAppCustomPermissionCache(context: Context) {
                     prefs.getString(miniAppId, ""),
                     object : TypeToken<MiniAppCustomPermission>() {}.type
                 )
-                val cachedPairs = cachedPermission.pairValues.toMutableList()
-
-                // detect any new change with comparing cached permissions and defaultDeniedList
-                // change means added new permission / removed existing permission from defaultDeniedList
-                val defaultPairs = defaultValue.pairValues
-                val changedPermissions = (defaultPairs + cachedPairs).groupBy { it.first.type }
-                    .filter { it.value.size == 1 }
-                    .flatMap { it.value }
-
-                return if (changedPermissions.isNotEmpty()) {
-                    if (cachedPairs.size < defaultPairs.size) {
-                        val filteredValue =
-                            MiniAppCustomPermission(miniAppId, cachedPairs + changedPermissions)
-                        applyStoringPermissions(filteredValue)
-                        filteredValue
-                    } else {
-                        cachedPairs.removeAll { (first) ->
-                            first.type in changedPermissions.groupBy { it.first.type }
-                        }
-                        val filteredValue = MiniAppCustomPermission(miniAppId, cachedPairs)
-                        applyStoringPermissions(filteredValue)
-                        filteredValue
-                    }
-                } else {
-                    val filteredValue = MiniAppCustomPermission(miniAppId, cachedPairs)
-                    applyStoringPermissions(filteredValue)
-                    filteredValue
-                }
+                cachedPermission
             } catch (e: Exception) {
                 // if there is any exception, just return the default value
                 defaultValue
```

Comment:
Is `defaultDeniedList` still correct? Shouldn't it return an empty list of permissions?

Reply:
Right, now we don't have to deal with any default list.